airflow/models.py

```diff
@@ -1898,6 +1900,7 @@ def __init__(
         self.params = params or {}  # Available in templates!
         self.adhoc = adhoc
         self.priority_weight = priority_weight
+        self.resources = resources or Resources()
```

Comment:
Perhaps this could be `self.resources = Resources(**(resources or {}))` to avoid the pain of having to import a Resource object. That way resources can always be expressed as a dict. The only requirement would be for Resource objects to comply with **splatting (I'm not sure how to do that offhand), I case the user passed a Resource instead of a dict.